### PR TITLE
prepare_sector_network: reorder buses for dac components

### PR DESCRIPTION
**Pull request by FabianHofmann**
Change the order of buses (bus0, bus1, ...) for DAC components to meet the convention of the other components: `bus0` refers to the electricity bus (input), `bus1` to the heat bus (input), 'bus2' to the CO2 atmosphere bus (input), and `bus3` to the CO2 storage bus (output).

Why is this helpful? 

For many routines in the pypsa statistics module, a convention where the `bus0` points to the location (in order to extract local balances) is assumed. With this change, the location of the DAC facilities is properly processed.  

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

File: scripts/prepare_sector_network.py [hunk 1]
```diff
@@ -710,27 +710,27 @@ def add_dac(n, costs):
...
+        bus2="co2 atmosphere",
+        bus3=spatial.co2.df.loc[locations, "nodes"].values,
         carrier="DAC",
         capital_cost=costs.at["direct air capture", "fixed"],
-        efficiency=1.0,
-        efficiency2=efficiency2,
-        efficiency3=efficiency3,
+        efficiency=-heat_input / electricity_input,
+        efficiency2=-electricity_input,
+        efficiency3=electricity_input,
```
Comment (fneum) [inline scripts/prepare_sector_network.py:733]: Shouldn't this be `1 / electricity_input`? The unit should be tCO2 / MWh_el.
```suggestion
        efficiency2=1 / -electricity_input,
        efficiency3=1 / electricity_input,
```

**Review by fneum — CHANGES_REQUESTED** [review]
I think there is one mistake with the units / efficiencies. Could you double check?

File: scripts/prepare_sector_network.py (hunk 1 above)
Comment (FabianHofmann) [reply scripts/prepare_sector_network.py]: you are totally right!

**Comment by FabianHofmann** [comment]
I made a small test case locally, this works as expected.

**Review by fneum — APPROVED** [review]

**Pull request merged**